### PR TITLE
add mouseDown event handler to trap click.

### PR DESCRIPTION
**Pull request by bobbyhubbard**
The focus was for some reason, causing the the mouseDown to be lost. Also fixed a few other errors

### CODE REVIEW

File: app/components/nag-es-autocomplete.js [hunk 1]
```diff
@@ -12,40 +12,42 @@ export default AutoComplete.extend({
...
-                }
+                },
+                "_source": ["org"],
+                "size": 10
             })
         });
         
         query.then(data => {
             Ember.run.later(() => {
-                var ids = Ember.Set.create();
```
Comment (bobbyhubbard) [inline app/components/nag-es-autocomplete.js]: Ember.Set deprecated

File: app/components/nag-es-autocomplete.js [hunk 2]
```diff
@@ -12,40 +12,42 @@ export default AutoComplete.extend({
...
             data: JSON.stringify({
-                "name-suggest": {
-                    "text": inputVal,
-                    "completion": {
-                        "field": "suggest",
-                        "size": 10,
-                        "fuzzy": {
-                            "fuzziness": 2
-                        }
+                "query": {
```
Comment (Blacktiger) [inline app/components/nag-es-autocomplete.js]: I wonder if we shouldn't look at using the official elasticsearch js client library? http://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/index.html

File: app/components/nag-es-autocomplete.js [hunk 3]
```diff
@@ -12,40 +12,42 @@ export default AutoComplete.extend({
...
             })
         });
         
         query.then(data => {
             Ember.run.later(() => {
-                var ids = Ember.Set.create();
-                var organizations = Ember.EnumerableUtils.filter(data['name-suggest'][0].options, option => {
-                    var id = option.payload.orgId;
-                    var isDuplicate = ids.contains(id);
+                var ids = new Set([]);
```
Comment (Blacktiger) [inline app/components/nag-es-autocomplete.js]: `new Set()` would avoid the additional array creation here I think

Comment (Blacktiger) [inline app/components/nag-es-autocomplete.js]: Yep, this is much better:

```
var organizations = new Set(data.hits.hits).map(hit => this.store.find('organizations/org', hit._source.org));
```

Comment (bobbyhubbard) [inline app/components/nag-es-autocomplete.js]: nice

Comment (bobbyhubbard) [inline app/components/nag-es-autocomplete.js]: set doesnt have a map

Comment (bobbyhubbard) [inline app/components/nag-es-autocomplete.js]: maybe entries

Comment (Blacktiger) [inline app/components/nag-es-autocomplete.js]: Maybe we can use `Ember.EnumerableUtils.map(new Set(data.hits.hits), hit => this.store.find('organizations/org', hit._source.org))`?

**Pull request merged**